django-stubs/utils/datastructures.pyi

```diff
@@ -83,11 +84,13 @@ class MultiValueDict(dict[_K, _V]):
     def __iter__(self) -> Iterator[_K]: ...
     # Fake to make `values` work properly
     def values(self) -> Iterator[_V | list[object]]: ...  # type: ignore[override]
+    def __copy__(self) -> MultiValueDict: ...
```

Comment:
No, it returns `Self` type: see https://github.com/django/django/blob/fcc5091b4a7c29e2912f6c8b2e0ebc64f4853457/django/utils/datastructures.py#L95-L105

Reply:
Fixed `Self` in `datastructures.py`, used the typing_extensions one good catch! Was there a specific reason to import `Self` from `_typeshed`, which is a TypeVar instead of `Self` from typing_extensions?

Reply:
Legacy stuff :)

Reply:
Okay, I will move the import to typing_extensions and remove redundant `self: Self` / `cls: type[Cls]` from this file then :)